nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -797,6 +799,140 @@ private void generateConstraintsForParam(
       argumentType =
           refineArgumentTypeWithDataflow(argumentType, argument, state, path, calledFromDataflow);
       solver.addSubtypeConstraint(argumentType, formalParamType, false);
+    } else {
+      LambdaExpressionTree lambda = (LambdaExpressionTree) argument;
+      handleLambdaArgumentInGenericMethodInference(
+          state, path, solver, allInvocations, formalParamType, calledFromDataflow, lambda);
+    }
+  }
+
+  private void handleLambdaArgumentInGenericMethodInference(
```

Comment:
Let's add Javadoc for this method.  The essence is that for lambdas, we add constraints based on any expressions returned from the lambda